### PR TITLE
Add eedit command to edit events

### LINKED ISSUE #135: 📖 As a returning user, I can change/edit the details of a birthday plan (event)
...so that PartyPlanet can reflect the changed plans.

________________________________________________________________
### Describe the feature
Add an eedit Command that will allow the user to edit the name/date/details of the Event
________________________________________________________________
### Additional context
*What is the current implementation, and the required scope of work?*

### PR DESCRIPTION
**Pull request by zhengruoxin**
Enables edit of event name, date and details. Closes #135.

Yet to add tests.

### CODE REVIEW

File: src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_REMARK;
+
+import seedu.partyplanet.commons.core.index.Index;
+import seedu.partyplanet.logic.commands.EEditCommand;
+import seedu.partyplanet.logic.commands.EEditCommand.EditEventDescriptor;
+import seedu.partyplanet.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new EditCommand object
```
Comment (Ellevy) [inline src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java:15]:
```suggestion
 * Parses input arguments and creates a new EEditCommand object
```

File: src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ */
+public class EEditCommandParser implements Parser<EEditCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditCommand
+     * and returns an EditCommand object for execution.
```
Comment (Ellevy) [inline src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java:21]:
```suggestion
     * Parses the given {@code String} of arguments in the context of the EEditCommand
     * and returns an EEditCommand object for execution.
```

File: src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+        model.setEvent(eventToEdit, editedEvent);
+        model.addState();
+        model.updateFilteredEventList(PREDICATE_SHOW_ALL_EVENTS);
+        return new CommandResult(String.format(MESSAGE_EDIT_EVENT_SUCCESS, editedEvent));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
```
Comment (Ellevy) [inline src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java:85]:
```suggestion
     * Creates and returns an {@code Event} with the details of {@code eventToEdit}
     * edited with {@code editEventDescriptor}.
```

File: src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+    public static final String MESSAGE_EDIT_EVENT_SUCCESS = "Edited event: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_EVENT = "This event already exists in PartyPlanet.";
+
+    private final Index index;
+    private final EditEventDescriptor editEventDescriptor;
+
+    /**
+     * @param index of the person in the filtered person list to edit
```
Comment (Ellevy) [inline src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java:48]:
```suggestion
     * @param index of the event in the filtered event list to edit
```

File: src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+import seedu.partyplanet.commons.util.CollectionUtil;
+import seedu.partyplanet.logic.commands.exceptions.CommandException;
+import seedu.partyplanet.model.Model;
+import seedu.partyplanet.model.event.Event;
+import seedu.partyplanet.model.event.EventDate;
+import seedu.partyplanet.model.person.Name;
+import seedu.partyplanet.model.person.Remark;
+
+/**
+ * Edits the details of an existing person in PartyPlanet.
```
Comment (Ellevy) [inline src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java:23]:
```suggestion
 * Edits the details of an existing event in PartyPlanet.
```

**Review by Ellevy — APPROVED** [review]
Other than minor javadocs changes, LGTM

**Review by garyljj — APPROVED** [review]
Other than stuff @Ellevy metioned, LGTM

File: src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java (hunk 1 above)
Comment (zhengruoxin) [reply src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java]: Adopted in latest commit.

File: src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java (hunk 2 above)
Comment (zhengruoxin) [reply src/main/java/seedu/partyplanet/logic/parser/EEditCommandParser.java]: Adopted in latest commit.

File: src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java (hunk 3 above)
Comment (zhengruoxin) [reply src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java]: Adopted in latest commit.

File: src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java (hunk 4 above)
Comment (zhengruoxin) [reply src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java]: Adopted in latest commit.

File: src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java (hunk 5 above)
Comment (zhengruoxin) [reply src/main/java/seedu/partyplanet/logic/commands/EEditCommand.java]: Adopted in latest commit.

**Pull request merged**